web/src/upload.js

```diff
@@ -311,7 +308,9 @@ export function setup_upload(config) {
     });
 
     uppy.on("upload-success", (file, response) => {
-        const url = response.body.uri;
+        const upload_url = response.uploadURL;
+        const upload_id = upload_url.split("/").at(-1);
+        const url = `/user_uploads/${page_params.realm_id}/${upload_id}/${file.name}`;
```

Comment:
Currently TusD doesn't support sending custom data in blocking hooks, So we will be having only file_id from TusD here. The URL calculation syncs with server. I have [talked](https://github.com/tus/tusd/issues/388#issuecomment-1588994095) with TusD maintainers, and they are hoping that TusD v2 (which will support sending custom data) will roll out roughly in 2 months. So after that we can do this in clean way, sending URL in the header from pre-finish hook.

Reply:
Cool, I posted on that issue asking if they have an update, since it'd be pretty exciting to be able to do this the clean way instead of having this API quirk and database migration.